python/cuxfilter/charts/deckgl/plots.py

```diff
@@ -242,3 +240,8 @@ def apply_theme(self, properties_dict):
                 "color_palette"
             ]
             self.compute_colors()
+        if self.map_style is None:
+            bg_color = properties_dict["map_style"]
+            self.chart._deck.map_style = (
+                f"mapbox://styles/mapbox/{bg_color}-v9"
```

Comment:
There appears to be a `dark-v10` if this link is accurate:

https://docs.mapbox.com/mapbox-gl-js/example/setstyle/

what happens if a users wants to use that? In general I think hardcoding the bits like `-v9` out of reach of users has the potential to frustrate. I would say it should always be accepted to specify any full mapbox name eg. `dark-v9` or `dark-v10` and if you want to have convenience aliases so that e.g. `foo` really means `foo-v9` that is fine (but should be explicitly communicated) 

I'm also confused about the link to the gallery in the doctoring above. "dark" does not appear anywhere on that page. Id "Decimal" (on that page) a valid value to pass as a style? (Also: does case matter?)

Reply:
actually that is just the theme default value, the user input for `map_style` is None, then map_style for the map is set as per the theme.


bg_color = "dark" for dark theme and rapids theme, and is "light" for light theme.


For the user input, we expect the full url as a string, like `mapbox://styles/mapbox/dark-v9`, or `mapbox://styles/mapbox/dark-v10`. Those urls work directly.

if None, pydeck defaults to its own default value, which as of version 0.5 is `mapbox://styles/mapbox/dark-v9`.

Reply:
Where/how is `properties_dict["map_style"]` set? Are it's only valid values `"light"` and `"dark"`? If so then I think the docstring above could reflect that the default map style is actually dependent on light vs dark, i.e. 

* `properties_dict["map_style"] == "light"` --> default is `mapbox://styles/mapbox/light-v9`
* `properties_dict["map_style"] == "dark"` --> default is `mapbox://styles/mapbox/dark-v9`

Is that correct?

That said, does this mean that the `map_style` parameter to `choropleth` and the `"map_style"` value in `properties_dict` can't both accept the same set of values?

Reply:
actually `properties_dict` is not exposed to user when intializing the chart. Its actually imported when a theme is applied,and is mentioned here - https://github.com/AjayThorve/cuxfilter/blob/patch_fixes/python/cuxfilter/themes/dark.py#L152,

https://github.com/AjayThorve/cuxfilter/blob/patch_fixes/python/cuxfilter/themes/light.py#L152

Reply:
so when you call `cux_df.dashboard([charts], theme=cuxfilter.themes.dark)`, all charts are intiailized, `chart.initiate_chart` is called, and all themes are applied, so `chart.apply_theme` is called before starting the dashboard

Reply:
I agree with you, the themes file should have the entire URL, instead of just the keywords `dark` and `light`

Reply:
In that case why not just put the real values `mapbox://styles/mapbox/dark-v9` and `mapbox://styles/mapbox/light-v9` in the dark/light them values so the f-string substitution can be removed (I just don't see the value of making that arbitrary transform in code if the theme can just have the correct value to begin)

Reply:
makes sense, I'll update the theme files with correct values

Reply:
> so when you call cux_df.dashboard([charts], theme=cuxfilter.themes.dark), all charts are intiailized

Ok, I see. Is the right POV to look at this then that the "dark" is the default theme? If that's the case then I think what's itching me is that individual defaults are specified "in place" as well as in the theme, and then if the theme ever changes it's hard to keep those in sync. Maybe the docs could delegate e.g. "The default value is the "dark" theme map style" then if the theme move to `-v10` (or whatever) then the docstrings are still true.

Reply:
fixed this, the default value will actually be:

```python
"""
default based on cuxfilter.themes:
            dark/rapids theme: 'mapbox://styles/mapbox/dark-v9'
            light theme: 'mapbox://styles/mapbox/light-v9'
"""
```